transports/quic/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+//! Implementation of the libp2p `Transport` and `StreamMuxer` traits for QUIC.
+//!
+//! # Usage
+//!
+//! Example:
+//!
+//! ```
+//! # #[cfg(not(feature = "async-std"))]
+//! # fn main() {}
+//! #
+//! # #[cfg(feature = "async-std")]
+//! # fn main() -> std::io::Result<()> {
+//! #
+//! use libp2p_quic as quic;
+//! use libp2p_core::{Multiaddr, Transport};
+//!
+//! let keypair = libp2p_core::identity::Keypair::generate_ed25519();
+//! let quic_config = quic::Config::new(&keypair).expect("could not make config");
+//!
+//! let mut quic_transport = quic::async_std::Transport::new(quic_config);
+//!
+//! let addr = "/ip4/127.0.0.1/udp/12345/quic".parse().expect("bad address?");
```

Comment:
nit: 
```suggestion
//! let addr = "/ip4/127.0.0.1/udp/12345/quic".parse().expect("transport should be able to listen on address");
```

https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style

Reply:
Done as part of fdca477c381a71cd8c52be341c3f74bdc8fd0198.